microscope/devices.py

```diff
@@ -667,18 +673,19 @@ def __init__(self, **kwargs):
                          self.get_roi,
                          self.set_roi,
                          None)
+
     def _process_data(self, data):
         """Apply self._transform to data."""
         flips = (self._transform[0], self._transform[1])
         rot = self._transform[2]
 
         # Choose appropriate transform based on (flips, rot).
-        return {(0, 0): numpy.rot90(data, rot),
-                (0, 1): numpy.flipud(numpy.rot90(data, rot)),
-                (1, 0): numpy.fliplr(numpy.rot90(data, rot)),
-                (1, 1): numpy.fliplr(numpy.flipud(numpy.rot90(data, rot)))
-                }[flips]
-
+        data = {(0, 0): numpy.rot90(data, rot),
+               (0, 1): numpy.flipud(numpy.rot90(data, rot)),
+               (1, 0): numpy.fliplr(numpy.rot90(data, rot)),
+               (1, 1): numpy.fliplr(numpy.flipud(numpy.rot90(data, rot)))
+               }[flips]
```

Comment:
Not introduced in this change but isn't this indexing of the dict always perfoming all flips and rotations and then we are discarding them all except one? If so, shouldn't we ensure that we only do the transformation that is actually required?

Reply:
Good point. I assumed lazy evaluation, but maybe that's not the case. Perhaps lambdas would be more appropriate.

Reply:
It's not evaluated lazily.  I'll wrap it with lambda and push a commit to the master branch.

Reply:
The fix is in this branch, because it'll introduce a merge conflict here if I push it to master.

Reply:
Fix included in this PR: cfff4a590a5b94b5aa5e680f05ebb6cf989d1bf0.